tests/test_utils.rs

```diff
@@ -37,3 +40,31 @@ pub fn continue_to_end(target: &LinuxTarget) {
         status => panic!("Status: {:?}", status),
     }
 }
+
+/// Turn the `pause` instruction inside the `breakpoint` function into a breakpoint.
+#[cfg(target_os = "linux")]
+#[allow(dead_code)]
+pub fn patch_breakpoint(target: &LinuxTarget, debuginfo: &RelocatedDwarf) {
+    // Test that `a_function` resolves to a function.
+    let breakpoint_addr = debuginfo.get_symbol_address("breakpoint").unwrap() + 4 /* prologue */;
```

Comment:
I guess function prologues have different sizes across different ISAs & platforms, so it might be worth it to put it as a const at the top of the file to emphasise this, e.g.:
```rust
/// Size of instructions in the function prologue
#[cfg(target_arch = "x86_64")]
pub const PROLOGUE_SIZE: usize = 4;
```
What do you think?

Reply:
This only works on x86_64 anyway. The test uses `core::arch::x86_64::_mm_pause()`. This is a temporary hack until we get real breakpoint support.

Reply:
Makes sense 👍